### ISSUE TITLE
Improve terminology and syntax around iterators

### ISSUE THREAD

**Issue by thallgren** [lyraproj/lyra#233]
The term "iteration" suggests that something happens in sequence but that is a false assumption in lyra since the default behavior for an "iteration" is to apply the activities in parallel. I would therefore suggest that we use the term "repeat" instead.

Another improvement is to combine the current "style" key (can be set to "each", "eachPair", "range", or "times") and the "over" key (holds the value), and rename the 'variable(s)' to 'as', such that:
```typescript
iteration: {
  style: 'each',
  over: [
    ['192.0.0.0', false, 'primary'],
    ['192.0.0.1', true, 'secondary']
  ],
  variables: ['ip', 'backup', 'role']
}

// Or for an example using times
iteration: {
  style: 'times',
  over: 5
}
```
becomes:
```typescript
repeat: {
  each: [
   ['192.0.0.0', false, 'primary'],
   ['192.0.0.1', true, 'secondary']
  ],
  as: [ip, backup, role]
}

// Or for an example using times
repeat: {
  times: 5
}
```

The YAML (where "style" is currently named "function", doh!) the equivalent would change from:
```YAML
iteration:
  function: each
  over:
    - ['192.0.0.0', false, primary]
    - ['192.0.0.1', true, secondary]
  variables: [ip, backup, role]

# Or for an example using times
iteration:
  function: times
  over: 5
```
into:
```YAML
repeat:
  each:
    - ['192.0.0.0', false, primary]
    - ['192.0.0.1', true, secondary]
  as: [ip, backup, role]

# Or for an example using times
repeat:
  times: 5
```

**Comment by thallgren** [comment]
The term `collect` came up as a replacement for `repeat`. It also resonates well with the term `into` which could be introduced to name the actual collection of outputs. Consider:
```typescript
collect: {
  each: [
   ['192.0.0.0', false, 'primary'],
   ['192.0.0.1', true, 'secondary']
  ],
  as: ['ip', 'replica', 'role'],
  into: 'loadBalancers'
}
```

**Comment by thallgren** [comment]
Closed by #252, lyraproj/puppet-workflow#20, and lyraproj/ts-workflow#5.
